### PR TITLE
Consolidates restricted visibility-based filtering

### PR DESCRIPTION
**Pull request by mvilanova**

### CODE REVIEW

File: src/dispatch/database/base.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import re
```
Comment (kevgliss) [inline src/dispatch/database/base.py:1]: nit: we seem to be using "base" a lot e.g. database.py import Base class now it's database.base import Base, this module might better named "core" or similar.

File: src/dispatch/database/service.py [hunk 2]
```diff
@@ -222,10 +153,26 @@ def search_filter_sort_paginate(
     else:
         query = db_session.query(model_cls)
 
+    if user_role != UserRoles.admin.value:
+        if model.lower() == "incident":
```
Comment (kevgliss) [inline src/dispatch/database/service.py:157]: Do we need to do something similar for tasks for restricted incidents?

**Review by kevgliss — APPROVED** [review]

File: src/dispatch/database/base.py (hunk 1 above)
Comment (mvilanova) [reply src/dispatch/database/base.py:1]: Yeah, sure. Honestly, I did not spend a lot of time thinking about it.

File: src/dispatch/database/service.py (hunk 2 above)
Comment (mvilanova) [reply src/dispatch/database/service.py:157]: I was going to add logic for it, but when I started testing tasks I realized they were already being filtered. I did some research to try to figure out the _why_ and the conclusion that I reached was that when we load `/tasks` we perform a GET against the `/api/v1/incidents/` endpoint, which returns a list of filtered incidents, and then sqlalchemy filters out the tasks based on that list.

**Pull request merged**